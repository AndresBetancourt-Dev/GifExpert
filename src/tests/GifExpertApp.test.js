import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Test of Component : <GifExpertApp/>", () => {
  test("GifExpertApp : Component Mounts Correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test("GifExpertApp : Component Mounts Correctly", () => {
    const categories = ["One Punch Man", "Goku"];
    const wrapper = shallow(<GifExpertApp initialState={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
