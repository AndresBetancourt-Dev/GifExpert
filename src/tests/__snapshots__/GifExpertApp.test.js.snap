// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Component : <GifExpertApp/> GifExpertApp : Component Mounts Correctly 1`] = `
<Fragment>
  <header
    className="animate__animated animate__fadeInTopLeft"
  >
    <img
      alt="Giphy"
      className="giphy"
      src="./images/giphy.png"
      title="Giphy"
    />
  </header>
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="One Punch Man"
    key="One Punch Man"
  />
</Fragment>
`;

exports[`Test of Component : <GifExpertApp/> GifExpertApp : Component Mounts Correctly 2`] = `
<Fragment>
  <header
    className="animate__animated animate__fadeInTopLeft"
  >
    <img
      alt="Giphy"
      className="giphy"
      src="./images/giphy.png"
      title="Giphy"
    />
  </header>
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="One Punch Man"
    key="One Punch Man"
  />
  <GifGrid
    category="Goku"
    key="Goku"
  />
</Fragment>
`;
